buildtools/commands/generate-phone-codes.js

```diff
@@ -0,0 +1,50 @@
+/* eslint no-eval: 0, max-len: 0, no-unused-vars: 0 */
+const { resolve } = require('path');
+const { writeFileSync } = require('fs');
+const axios = require('axios');
+
+const METADATA_URI = 'https://raw.githubusercontent.com/googlei18n/libphonenumber/master/javascript/i18n/phonenumbers/metadata.js';
+const ROOT_DIR = resolve(__dirname, '../../');
+const OUTPUT_FILE = resolve(ROOT_DIR, 'src/util/countryCallingCodes.js');
+
+exports.command = 'generate-phone-codes';
+exports.describe = 'Downloads latest metadata from google libphonenumber and generates ' +
+  'json we use to map region codes to country calling codes';
+exports.handler = () => {
+  console.log('Downloading latest metadata from google libphonenumber...');
+  axios.get(METADATA_URI)
+    .then(function ({ data }) {
+      const regionCodesToCallingCodeMap = {};
+      const goog = { provide: function () { } };
+      const i18n = { phonenumbers: { metadata: {} } };
+      eval(data);
+      Object.keys(i18n.phonenumbers.metadata.countryCodeToRegionCodeMap)
+        .forEach(function (callingCode) {
+          const regionCodes = i18n.phonenumbers.metadata.countryCodeToRegionCodeMap[callingCode];
+          regionCodes.forEach(function (regionCode) {
+            regionCodesToCallingCodeMap[regionCode] = Number(callingCode);
+          });
+        });
+
+      // Add in extra codes that are not in the metadata but appear in
+      // our country.properties list. Sourced from:
+      // https://en.wikipedia.org/wiki/List_of_country_calling_codes
+      // http://www.auslandsvorwahlen.net/en/
+      // http://www.listofcountriesoftheworld.com/country-dialling-codes.html
+      // Note: TF, BV, and HM do not have phone codes
+      regionCodesToCallingCodeMap['GS'] = 500;
+      regionCodesToCallingCodeMap['PN'] = 64;
+      regionCodesToCallingCodeMap['AQ'] = 672;
+      regionCodesToCallingCodeMap['UM'] = 1;
+      regionCodesToCallingCodeMap['AN'] = 599;
+
+      const contents = 'define(' + JSON.stringify(regionCodesToCallingCodeMap, null, 2) + ')';
+      writeFileSync(OUTPUT_FILE, contents);
```

Comment:
Doesn't have to be today, but we should probably change how this currently works.

We're currently fetching some URL from `master` (should be tied to a SHA instead), and then writing to a file once the phone codes have been generated. Ideally we will have an **update** task that can be run to assert that the checked in Google lib matches our current `countryCallingCodes.js` file, instead of doing this every time `yarn start` is run.

Maybe we update the description of OKTA-198793 to include changing behavior too?

Reply:
- why should be tied to SHA instead of master?
- what's the advantage to have an *update* task to do assertion (I'll the *update* task would also need to download file and parse it) rather than generating every time?